src/main/java/com/wire/kalium/api/assets/AssetsApi.kt

```diff
@@ -0,0 +1,12 @@
+package com.wire.kalium.api.assets
+
+import com.wire.kalium.api.KaliumHttpResult
+import io.ktor.client.statement.*
+
+interface AssetsApi {
+
+    suspend fun downloadAsset(assetKey: String, assetToken: String?): KaliumHttpResult<HttpResponse>
```

Comment:
Why wrap a `HttpResponse`?

Shouldn't it be something like `KaliumHttpResult<Asset>` where Asset contains mimetype and byte array, or similar?

Passing the raw HTTP result forward and exposing Ktor to whoever uses the lib is probably not wanted.

Reply:
A more philosophical question is:

Suppose the asset is an image, how do users of this library can use it?

Something along these lines maybe? 

```kotlin
val asset = api.downloadAsset(...).assetStuff

asset.data // bytearray
asset.type // mpeg, jpeg, gif, png, pdf
```

Reply:
Yes you are right about the lack of specification of the response object type. However this was intended as the documentation in Swagger from this endpoint is incomplete and I couldn't get any insightful info from the Backend team. So I decided to leave it as an HttpResponse object in case we want to debug it and access to the information manually from the `CLIApplication` client

Reply:
So I guess this can still be considered as WIP until we make more sense out of the Swagger docs or the backend team replies back with a thorough explanation on how to execute this endpoint and what kind of object we should expect from it on the response